packages/field/src/react/__stories__/tags-input-field.story.tsx

```diff
@@ -0,0 +1,250 @@
+              >
+                {option.label}
+              </TagsFieldTag>
+            )
+          })}
+
+          <TagsFieldInput
+            id={inputId}
+            onChange={handleFilterTermChange}
+            value={filterTerm}
+            {...getDropdownProps()}
+          />
+        </div>
+      </Field>
+
+      <div
+        style={{
+          border: '2px dashed pink',
+          margin: '20px 0',
+          maxHeight: 200,
+          overflow: 'scroll',
+          padding: 20
+        }}
+      >
+        <ul>
+          {filterResults.map((option, index) => (
+            <li key={`filter-result-${index}`}>
+              <span>{option.label} </span>
+              <button onClick={e => handleAddSelected(e, option.value)}>
+                add
+              </button>
+            </li>
+          ))}
+        </ul>
+      </div>
+    </>
+  )
+}
+
+const CustomFieldTag = forwardRef<HTMLDivElement, HTMLPropsFor<'div'>>(
+  (props, ref) => <div ref={ref} {...props} {...css({ padding: 0 })} />
+)
+
+interface TagsFieldTagProps extends HTMLPropsFor<'div'> {
+  onRequestRemove: MouseEventHandler
+}
+const TagsFieldTag = forwardRef<HTMLDivElement, TagsFieldTagProps>(
+  (props, ref) => {
+    const { children, onRequestRemove, ...rest } = props
+
+    return (
+      <div
+        ref={ref}
+        {...rest}
+        {...css({ margin: `calc(${GUTTER_SIZE}px / 2)` })}
+      >
+        {/*
+          NOTE: Using isPressed prop to get blue background...
+                Tag should have additional variants if this is something we want
+                to officially support
+        */}
+        <Tag
+          icon={<CloseIcon onClick={onRequestRemove} />}
+          isPressed
+          size={Tag.sizes.small}
+        >
+          {children}
+        </Tag>
+      </div>
+    )
+  }
+)
+
+const TagsFieldInputContainer = forwardRef<HTMLDivElement, HTMLPropsFor<'div'>>(
+  (props, ref) => (
+    <div
+      ref={ref}
+      {...props}
+      {...css({ margin: `calc(${GUTTER_SIZE}px / 2)` })}
+    />
+  )
+)
+interface TagsFieldInputProps extends ComponentProps<typeof Field.Input> {}
+const TagsFieldInput = forwardRef<HTMLInputElement, TagsFieldInputProps>(
+  (props, ref) => {
+    return (
+      <Field.Input
+        renderContainer={TagsFieldInputContainer}
+        ref={ref}
+        type="text"
+        {...props}
+        {...css({ minWidth: 50 })}
+      />
+    )
+  }
+)
+
+const ConstrainWidthDecorator = (Story: Story) => {
+  return (
+    <div style={{ maxWidth: '400px' }}>
+      <Story />
+    </div>
+  )
+}
+
+export default {
+  title: 'Components/Field/TagsField',
+  component: TagsField,
+  decorators: [ConstrainWidthDecorator],
+  args: { options: periodicElements }
+} as Meta
+
+const Template: Story<ComponentProps<typeof TagsField>> = args => {
+  return <TagsField {...args} />
+}
+
+export const Basic = Template.bind({})
+
+export const WithSuffix = Template.bind({})
+WithSuffix.args = {
```

Comment:
Why is this done?  Is the `.args` thing storybook-specific?

Reply:
yes it's a storybook think. you define default props and bind a Template. then you can pass in new args for each additional story you create